packages/block-editor/src/components/global-styles/color-panel.js

```diff
@@ -0,0 +1,754 @@
+					userValue: userGradient,
+					isGradient: true,
+				},
+			],
+		},
+		showLinkPanel && {
+			key: 'link',
+			label: __( 'Link' ),
+			hasValue: hasLink,
+			resetValue: resetLink,
+			isShownByDefault: defaultControls.link,
+			indicators: [ linkColor, hoverLinkColor ],
+			tabs: [
+				{
+					key: 'link',
+					label: __( 'Default' ),
+					inheritedValue: linkColor,
+					setValue: setLinkColor,
+					userValue: userLinkColor,
+				},
+				{
+					key: 'hover',
+					label: __( 'Hover' ),
+					inheritedValue: hoverLinkColor,
+					setValue: setHoverLinkColor,
+					userValue: userHoverLinkColor,
+				},
+			],
+		},
+	].filter( Boolean );
+
+	elements.forEach( ( { name, label, showPanel } ) => {
+		if ( ! showPanel ) return;
+
+		const elementBackgroundColor = decodeValue(
+			inheritedValue?.elements?.[ name ]?.color?.background
+		);
+		const elementGradient = decodeValue(
+			inheritedValue?.elements?.[ name ]?.color?.gradient
+		);
+		const elementTextColor = decodeValue(
+			inheritedValue?.elements?.[ name ]?.color?.text
+		);
+		const elementBackgroundUserColor = decodeValue(
+			value?.elements?.[ name ]?.color?.background
+		);
+		const elementGradientUserColor = decodeValue(
+			value?.elements?.[ name ]?.color?.gradient
+		);
+		const elementTextUserColor = decodeValue(
+			value?.elements?.[ name ]?.color?.text
+		);
+		const hasElement = () =>
+			!! (
+				elementTextUserColor ||
+				elementBackgroundUserColor ||
+				elementGradientUserColor
+			);
+		const resetElement = () => {
+			onChange( {
+				...value,
+				elements: {
+					...value?.elements,
+					[ name ]: {
+						...value?.elements?.[ name ],
+						color: {
+							...value?.elements?.[ name ]?.color,
+							background: undefined,
+							gradient: undefined,
+							text: undefined,
+						},
+					},
+				},
+			} );
+		};
+
+		const setElementTextColor = ( newValue ) => {
+			onChange( {
+				...value,
+				elements: {
+					...value?.elements,
+					[ name ]: {
+						...value?.elements?.[ name ],
+						color: {
+							...value?.elements?.[ name ]?.color,
+							text: encodeColorValue( newValue ),
+						},
+					},
+				},
+			} );
+		};
+		const setElementBackgroundColor = ( newValue ) => {
+			onChange( {
+				...value,
+				elements: {
+					...value?.elements,
+					[ name ]: {
+						...value?.elements?.[ name ],
+						color: {
+							...value?.elements?.[ name ]?.color,
+							background: encodeColorValue( newValue ),
+							gradient: undefined,
+						},
+					},
+				},
+			} );
+		};
+		const setElementGradient = ( newValue ) => {
+			onChange( {
+				...value,
+				elements: {
+					...value?.elements,
+					[ name ]: {
+						...value?.elements?.[ name ],
+						color: {
+							...value?.elements?.[ name ]?.color,
+							gradient: encodeGradientValue( newValue ),
+							background: undefined,
+						},
+					},
+				},
+			} );
+		};
+		const supportsTextColor = true;
+		const supportsBackground = name !== 'caption';
```

Comment:
Is it worth adding a comment about why this is a one-off for `caption`? I could imagine this being tricky to find, or potentially confusing for someone else looking at this. From memory, background color support for `caption` is disabled because it doesn't have padding support?

```suggestion
		// Background color is not supported for `caption` as there isn't yet a way to set padding for the element.
		const supportsBackground = name !== 'caption';
```